modules/web/web-impl/src/main/java/com/enonic/xp/web/impl/context/ContextFilter.java

```diff
@@ -30,6 +32,12 @@ protected void doHandle( final HttpServletRequest req, final HttpServletResponse
         context.getLocalScope().setAttribute( "__currentTimeMillis", System.currentTimeMillis() );
         context.getLocalScope().setSession( new SessionWrapper( req ) );
 
+        final VirtualHost virtualHost = VirtualHostHelper.getVirtualHost( req );
+        if ( virtualHost != null )
+        {
+            virtualHost.getContext().forEach( ( key, value ) -> context.getLocalScope().setAttribute( key, value ) );
```

Comment:
One annoying thing,  context contains typed values for branch, repository, etc. 
They have to be converted

Reply:
How that is related with my changes?

Reply:
This pr introduces vhost context feature.
Context supports default repositoryId, branch and authenticationInfo - but it is not possible to set them up using just string.

AuthenticationInfo is probably too complex, but repositoryid and branch should be supported

Reply:
Ok, I agree. But I don't want to do it in this PR. I suggest having a separate issue for that because it will be more transparent, and it requires more work than one might think at first.